content/blog/ssh_port_forwarding.md

```diff
@@ -0,0 +1,224 @@
+
+The beauty of SSH port forwarding lies in its simplicity and versatility. You're essentially borrowing the SSH connection to carry other types of traffic, all while maintaining the security and encryption that SSH provides. This makes it perfect for accessing services that might be blocked, restricted, or simply need an extra layer of security.
+
+## The Three Types of SSH Port Forwarding
+
+### Local Port Forwarding (Most Common)
+
+Local port forwarding is probably what you'll use 90% of the time. It forwards a port on your local machine to a port on a remote machine through the SSH connection. This is incredibly useful when you need to access a service running on a remote server that's not directly accessible from your location.
+
+The syntax comes in short and long forms:
+
+**Short form:**
+```bash
+ssh -L 8080:localhost:80 user@server
+```
+
+**Long form:**
+```bash
+ssh -L localhost:8080:localhost:80 user@server
+```
+
+
+{{< image "ssh_port_forwarding/local_port_forwarding.webp" "Local Port Forwarding Example" >}}
+
+Here's a practical example. Let's say you're working from a coffee shop and need to access your company's database server that's only accessible from within the office network. You can create a tunnel through your office SSH server:
+
+```bash
+ssh -L 5432:db.company.com:5432 john@office-server.company.com
+```
+
+Now you can connect to `localhost:5432` on your laptop, and it'll be as if you're connecting directly to the database from inside the office network. Your database client doesn't know the difference, but all the traffic is securely encrypted through the SSH tunnel.
+
+### Remote Port Forwarding
+
+Remote port forwarding works in the opposite direction. It forwards a port on the remote server back to your local machine. This is super handy when you're developing something locally and need to share it with someone else, or when you need to expose a local service to a remote network. For a deeper dive into remote tunneling, check out our {{< link href="https://pinggy.io/blog/ssh_reverse_tunnelling/" >}}SSH reverse tunneling guide{{< /link >}}.
+
+```bash
+ssh -R 0.0.0.0:8080:server:80 user@gateway
+```
+
+{{< image "ssh_port_forwarding/remote_port_forwarding.webp" "Local Port Forwarding Example" >}}
+
+Imagine you're building a webhook handler for a payment processor, but you're developing it locally. The payment service needs to send webhooks to your application, but it can't reach your laptop directly. Remote port forwarding to the rescue:
+
+```bash
+ssh -R 0.0.0.0:8080:server:80 john@my-public-server.com
+```
+
+Now when the payment service sends a webhook to `my-public-server.com:8080`, it gets forwarded to your local development server running on port 3000. You can test webhooks in real-time without deploying anything.
```

Comment:
this does not match with the above command. then change the port 80 to 3000 in the above command.